vowpalwabbit/core/src/reductions/search/search.cc

```diff
@@ -2775,6 +2775,13 @@ void search::set_label_parser(VW::label_parser& lp, bool (*is_test)(const VW::po
   {
     priv->all->logger.err_warn("Task should not set label parser except in initialize function.");
   }
+
+  // This will override the label parser object passed in, and it will persist.
+  // I don't know why Search needs to override is_test, and if it didn't then this would not be necessary.
+  // Remove the overriding of is_test breaks the unit tests.
+  // TODO: figure out why Search needs to override is_test and remove this.
+  lp.test_label = is_test;
```

Comment:
This is a hack for the fact search overrides the is_test function for label parsers